File: src/Entity/Chambre.php
<?php

namespace App\Entity;

use App\Repository\ChambreRepository;
<<<<<<< Updated upstream
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
=======
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
use Symfony\Component\Serializer\Annotation\Groups;
>>>>>>> Stashed changes

/**
 * @ORM\Entity(repositoryClass=ChambreRepository::class)
 */
class Chambre
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @ORM\Column(type="integer")
     * @Assert\NotBlank(message="le numéro de la chambre est obligatoire")
<<<<<<< Updated upstream
     *      @Assert\Length(
     *      max=20,
     *      maxMessage  ="Numéro chambre ne doit pas depasser 20 chiffres ",
     * )
=======

>>>>>>> Stashed changes
     */
    private $NumCh;

    /**
     * @ORM\Column(type="integer")
     * @Assert\NotBlank(message="le nombre de lits de la chambre est obligatoire")
     */
    private $NbrLits;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="Ce champ est obligatoire")
     */
    private $vue;

    /**
     * @ORM\Column(type="integer")
     * @Assert\NotBlank(message="Ce champ est obligatoire")
     */
    private $Etage;

    /**
     * @ORM\Column(type="float")
     * @Assert\NotBlank(message="le prix de la chambre est obligatoire")
     * @Assert\GreaterThanOrEqual(
<<<<<<< Updated upstream
     *     value = 150,message="Le prix minimale de la chambre est  150Dt"
=======
     *     value = 50,message="Le prix minimale de la chambre est  150Dt"
>>>>>>> Stashed changes
     * )
     * @Assert\LessThanOrEqual(
     *     value = 2000,message="Le prix de la chambre ne doit pas depasser 2000Dt"
     * )
     */
    private $Prix;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="Ce champ est obligatoire")
     */
    private $Bloc;

<<<<<<< Updated upstream
=======
    /**
     * @ORM\Column(type="string", length=255)
     */
    private $dispo;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="ajouter une image")
     */
    private $ImageCh;

    /**
     * @ORM\ManyToOne(targetEntity=Hotel::class, inversedBy="chambre")
     */
    private $hotel;

    /**
     * @ORM\OneToMany(targetEntity=ResChambre::class, mappedBy="chambre",cascade={"remove"})
     */
    private $resChambre;

    public function __construct()
    {
        $this->resChambres = new ArrayCollection();
        $this->resChambre = new ArrayCollection();
    }

>>>>>>> Stashed changes
    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNumCh(): ?int
    {
        return $this->NumCh;
    }

    public function setNumCh(int $NumCh): self
    {
        $this->NumCh = $NumCh;

        return $this;
    }

    public function getNbrLits(): ?int
    {
        return $this->NbrLits;
    }

    public function setNbrLits(int $NbrLits): self
    {
        $this->NbrLits = $NbrLits;

        return $this;
    }

    public function getVue(): ?string
    {
        return $this->vue;
    }

    public function setVue(string $vue): self
    {
        $this->vue = $vue;

        return $this;
    }

    public function getEtage(): ?int
    {
        return $this->Etage;
    }

    public function setEtage(int $Etage): self
    {
        $this->Etage = $Etage;

        return $this;
    }

    public function getPrix(): ?float
    {
        return $this->Prix;
    }

    public function setPrix(float $Prix): self
    {
        $this->Prix = $Prix;

        return $this;
    }

    public function getBloc(): ?string
    {
        return $this->Bloc;
    }

    public function setBloc(string $Bloc): self
    {
        $this->Bloc = $Bloc;

        return $this;
    }
<<<<<<< Updated upstream
=======

    public function getDispo(): ?string
    {
        return $this->dispo;
    }

    public function setDispo(string $dispo): self
    {
        $this->dispo = $dispo;

        return $this;
    }

    public function getImageCh()
    {
        return $this->ImageCh;
    }

    public function setImageCh( $ImageCh)
    {
        $this->ImageCh = $ImageCh;

        return $this;
    }

    public function getHotel(): ?Hotel
    {
        return $this->hotel;
    }

    public function setHotel(?Hotel $hotel): self
    {
        $this->hotel = $hotel;

        return $this;
    }

    /**
     * @return Collection<int, ResChambre>
     */
    public function getResChambres(): Collection
    {
        return $this->resChambres;
    }

    public function addResChambre(ResChambre $resChambre): self
    {
        if (!$this->resChambres->contains($resChambre)) {
            $this->resChambres[] = $resChambre;
            $resChambre->setChambre($this);
        }

        return $this;
    }

    public function removeResChambre(ResChambre $resChambre): self
    {
        if ($this->resChambres->removeElement($resChambre)) {
            // set the owning side to null (unless already changed)
            if ($resChambre->getChambre() === $this) {
                $resChambre->setChambre(null);
            }
        }

        return $this;
    }

    public function __toString()
    {
        return(string)$this->getNumCh();
    }

    /**
     * @return Collection<int, ResChambre>
     */
    public function getResChambre(): Collection
    {
        return $this->resChambre;
    }

>>>>>>> Stashed changes
}

File: src/entities/Chambre.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package entities;

import java.util.Objects;

/**
 *
 * @author 123
 */
public class Chambre {
    private int id,NbrLits,NumCh,Etage ;
    private String vue,Bloc,dispo,ImageCh;
    private float Prix;
    private int hotel_id;

    public Chambre() {
    }

    public Chambre(int NbrLits, int NumCh, int Etage, String vue, String Bloc, String dispo, String ImageCh, float Prix, int hotel_id) {
        this.NbrLits = NbrLits;
        this.NumCh = NumCh;
        this.Etage = Etage;
        this.vue = vue;
        this.Bloc = Bloc;
        this.dispo = dispo;
        this.ImageCh = ImageCh;
        this.Prix = Prix;
        this.hotel_id = hotel_id;
    }

    public Chambre(int id, int NbrLits, int NumCh, int Etage, String vue, String Bloc, String dispo, String ImageCh, float Prix) {
        this.id = id;
        this.NbrLits = NbrLits;
        this.NumCh = NumCh;
        this.Etage = Etage;
        this.vue = vue;
        this.Bloc = Bloc;
        this.dispo = dispo;
        this.ImageCh = ImageCh;
        this.Prix = Prix;
    }

    public Chambre(int NbrLits, int NumCh, int Etage, String vue, String Bloc, String dispo, String ImageCh, float Prix) {
        this.NbrLits = NbrLits;
        this.NumCh = NumCh;
        this.Etage = Etage;
        this.vue = vue;
        this.Bloc = Bloc;
        this.dispo = dispo;
        this.ImageCh = ImageCh;
        this.Prix = Prix;
    }

    public Chambre(int id, int NbrLits, int NumCh, int Etage, String vue, String Bloc, String dispo, String ImageCh, float Prix, int hotel_id) {
        this.id = id;
        this.NbrLits = NbrLits;
        this.NumCh = NumCh;
        this.Etage = Etage;
        this.vue = vue;
        this.Bloc = Bloc;
        this.dispo = dispo;
        this.ImageCh = ImageCh;
        this.Prix = Prix;
        this.hotel_id = hotel_id;
    }

    
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getNbrLits() {
        return NbrLits;
    }

    public void setNbrLits(int NbrLits) {
        this.NbrLits = NbrLits;
    }

    public int getNumCh() {
        return NumCh;
    }

    public void setNumCh(int NumCh) {
        this.NumCh = NumCh;
    }

    public int getEtage() {
        return Etage;
    }

    public void setEtage(int Etage) {
        this.Etage = Etage;
    }

    public String getVue() {
        return vue;
    }

    public void setVue(String vue) {
        this.vue = vue;
    }

    public String getBloc() {
        return Bloc;
    }

    public void setBloc(String Bloc) {
        this.Bloc = Bloc;
    }

    public String getDispo() {
        return dispo;
    }

    public void setDispo(String dispo) {
        this.dispo = dispo;
    }

    public String getImageCh() {
        return ImageCh;
    }

    public void setImageCh(String ImageCh) {
        this.ImageCh = ImageCh;
    }

    public float getPrix() {
        return Prix;
    }

    public void setPrix(float Prix) {
        this.Prix = Prix;
    }

    public int getHotel_id() {
        return hotel_id;
    }

    public void setHotel_id(int hotel_id) {
        this.hotel_id = hotel_id;
    }

    @Override
    public String toString() {
        return "Chambre{" + "id=" + id + ", NbrLits=" + NbrLits + ", NumCh=" + NumCh + ", Etage=" + Etage + ", vue=" + vue + ", Bloc=" + Bloc + ", dispo=" + dispo + ", ImageCh=" + ImageCh + ", Prix=" + Prix + ", hotel_id=" + hotel_id + '}';
    }

    @Override
    public int hashCode() {
        int hash = 3;
        return hash;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        final Chambre other = (Chambre) obj;
        if (this.NbrLits != other.NbrLits) {
            return false;
        }
        if (this.NumCh != other.NumCh) {
            return false;
        }
        if (this.Etage != other.Etage) {
            return false;
        }
        if (Float.floatToIntBits(this.Prix) != Float.floatToIntBits(other.Prix)) {
            return false;
        }
        if (this.hotel_id != other.hotel_id) {
            return false;
        }
        if (!Objects.equals(this.vue, other.vue)) {
            return false;
        }
        if (!Objects.equals(this.Bloc, other.Bloc)) {
            return false;
        }
        if (!Objects.equals(this.dispo, other.dispo)) {
            return false;
        }
        if (!Objects.equals(this.ImageCh, other.ImageCh)) {
            return false;
        }
        return true;
    }


    
    
}
